src/catalog/src/builtin.rs

```diff
@@ -6127,13 +6126,13 @@ pub static MZ_MAPPABLE_OBJECTS: LazyLock<BuiltinView> = LazyLock::new(|| {
         ("global_id", "The global ID of the object."),
     ]),
     sql: "
-SELECT SUBSTRING(name FROM 11) AS name, global_id
-FROM mz_introspection.mz_dataflows md JOIN mz_introspection.mz_dataflow_global_ids mdgi USING (id)
-WHERE name LIKE 'Dataflow: %' AND
-      (global_id IN (SELECT id FROM mz_catalog.mz_indexes UNION SELECT on_id FROM mz_catalog.mz_indexes)
-       OR EXISTS (SELECT 1 FROM mz_catalog.mz_materialized_views mmv WHERE POSITION(mmv.name IN name) <> 0))",
+SELECT md.name || '.' || ms.name || '.' || mo.name AS name, mgi.global_id AS global_id
```

Comment:
We should pass these names through `quote_ident` to make sure we quote them correctly if they contain weird characters.

```suggestion
SELECT quote_ident(md.name) || '.' || quote_ident(ms.name) || '.' || quote_ident(mo.name) AS name, mgi.global_id AS global_id
```